ghga_service_chassis_lib/s3.py

```diff
@@ -172,6 +172,46 @@ def _translate_s3_client_errors(
     return exception
 
 
+def adapt_part_size(current_part_size: int, file_size: int) -> int:
+    """
+    Checks if the provided part size is within bounds and produces <= 10_000 parts.
+
+
+    Args:
+        current_part_size (int): currently chosen part size in bytes to be checked
+        file_size (int): total file size in bytes
+
+    Raises:
+        ValueError if file size exceeds the maximum of 5 TiB
+    """
```

Comment:
This is great. But could we make the current_part_size optional, so that you can give your desired part size but don't have to?

Something in the direction of: 
```suggestion
def calc_part_size(*, file_size: int, preferred_part_size: Optional[int] = None) -> int:
    """
    Gives recommendations on the part_size to use for up- or download of a file given 
    it's total size. It makes sure that chosen part size is within bounds and produces 
    <= 10_000 parts.

    Args:
        file_size (int): total file size in bytes
        preferred_part_size (int):
            You may provide your preferred part size in bytes. If it satisfies the technical
            constraints, it will be returned unchanged. Otherwise, it is ignored and a 
            technically viable part size is chosen instead.
            
    Returns: a recommendation for the part size in bytes.
        
    Raises:
        ValueError if file size exceeds the maximum of 5 TiB
    """
    
    if preferred_part_size is None:
        preferred_part_size = 8 * 1024**2
        
```